tests/testthat/test-integration_metrics.R

```diff
@@ -0,0 +1,109 @@
+# generate testing data
+set.seed(1665)
+merged_sce <- scpcaTools:::sim_sce(n_cells = 300, n_genes = 100, n_empty = 0)
+batches <- rep(c("a", "b", "c"), each = 100)
+barcodes <- rownames(colData(merged_sce))
+
+# set up colData rownames as batch-barcode
+new_rownames <- glue::glue("{batches}-{barcodes}")
+rownames(colData(merged_sce)) <- new_rownames
+
+# Add "sample" to colData
+colData(merged_sce)$sample <- batches
+
+# add PCs for testing (exact numbers don't matter)
+# make a 300x100 matrix
+reducedDim(merged_sce, "PCA") <- matrix(runif(300 * 100, min = 0, max = 100), nrow = 300)
```

Comment:
I wonder if we might set these up to get specific ASW values? Probably not necessary.